sources/core/Xenko.Core.Mathematics/Vector3.cs

```diff
@@ -144,6 +144,17 @@ public Vector3(float[] values)
             Z = values[2];
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Xenko.Core.Mathematics.Vector3"/> struct.
+        /// </summary>
+        /// <param name="v">The Double3 to construct the Vector3 from.</param>
+        public Vector3(Double3 v)
```

Comment:
I'm not a fan of constructors that might cause runtime exception because of the downcast from `double`to `float`.
I would prefer to have an explicit operator:
```csharp
public static explicit operator Vector3(Double3 value)
{
    return new Vector3((float)value.X, (float)value.Y, (float)value.Z);
}
```

Reply:
I wasn't aware that it was possible for that to cause a runtime exception? Are there any particular bad values you have in mind which would cause this?

Reply:
My bad. I thought it would throw an `OverflowException` if you had something like `float f = (float)(((double)float.MaxValue)*2)`. But apparently, that's only the case for signed/unsigned pair for integral types.

But as @xen2 said, when losing precision explicit cast is preferred anyway.

Reply:
This, and the licensing stuff, have now been changed.